### ISSUE TITLE
Remove "display notifications" permission when it isn't needed

### ISSUE THREAD

**Issue by DNin01** [ScratchAddons/ScratchAddons#6863]
### Why this would be helpful

Currently, after a user enables the Scratch Notifier addon and gives the extension permission to display notifications, they can't take it back. They can disable the addon, but AFAIK, there is no way for the user to revoke an optional permission they have already granted an extension. However, extensions _can_ [remove optional permissions](https://developer.chrome.com/docs/extensions/reference/permissions/#step-5-remove-the-permissions). We could technically do that when the addon is disabled.

Note that if the user has granted the extension an optional permission and then that permission is later removed, if the permission is ever requested again, Chrome may not ask the user and instead immediately grant the extension permission, so it doesn't truly revoke the ability to display notifications. Removing the permission does, however, mean that we have to get the permission again before we can display notifications again.

### My suggestion

Remove the optional `notifications` permission when the `scratch-notifier` addon is disabled, or after some time of it being, and I think `clipboardWrite` with some other addons on Firefox (if I'm correct? And if it's even possible).

### Additional context

Related:
- #6401

**Comment by WorldLanguages** [comment]
> Note that if the user has granted the extension an optional permission and then that permission is later removed, if the permission is ever requested again, Chrome may not ask the user and instead immediately grant the extension permission, so it doesn't truly revoke the ability to display notifications. Removing the permission does, however, mean that we have to get the permission again before we can display notifications again.

Sorry, I had a hard time reading this. Could you clarify what the goal is here?
Are you saying that -removing a permission and requesting it again- does not open a prompt and instead the browser already grants it? If so, what's the point of revoking the permission?

**Comment by DNin01** [comment]
> > Note that if the user has granted the extension an optional permission and then that permission is later removed, if the permission is ever requested again, Chrome may not ask the user and instead immediately grant the extension permission, so it doesn't truly revoke the ability to display notifications. Removing the permission does, however, mean that we have to get the permission again before we can display notifications again.
> 
> 
> 
> Sorry, I had a hard time reading this. Could you clarify what the goal is here?
> 
> Are you saying that -removing a permission and requesting it again- does not open a prompt and instead the browser already grants it?

Yes, that's correct. The documentation said that is _usually_ the case.

> If so, what's the point of revoking the permission?

Yeah, really. Probably mostly just so the user can be sure that there's no way for us to exploit that permission in the background (including by accident), which sounds kind of silly to me when you can just get it back without actual permission from the user. But Google's documentation recommends removing unneeded permissions and that's why I suggested it.

**Comment by WorldLanguages** [comment]
Okay, so this is just a best practice.

**Comment by DNin01** [comment]
Here's an unmentioned reason I suggested this: the possibility of using the permission to store the enabled state of `scratch-notifier`. So, the addon's state would not be stored in `addonsEnabled` in extension storage, but instead, the status of the `"notifications"` permission—whether it is granted or not—would indicate if the addon is supposed to be in an enabled or disabled state.

Permissions are not synced between devices, so this would be a local state.

**Comment by WorldLanguages** [comment]
Good point - sync doesn't consider that permissions may be different between devices.

However more addons in the future may request the same permission, so I would suggest against using a permission as an indicator of whether a specific addon is enabled
